src/analyze/analyze-verify.c

```diff
@@ -233,6 +248,16 @@ int verify_units(char **filenames, UnitFileScope scope, bool check_man, bool run
         if (strv_isempty(filenames))
                 return 0;
 
+        /* Allow systemd-analyze to hook in a callback function so that it can get
+         * all the required log data from the function itself without having to rely
+         * on a global set variable for the same */
+        if (return_error_on != RETURN_ERROR_ON_NONE) {
+                s = set_new(&string_hash_ops);
+                if (!s)
+                        return -ENOMEM;
+                set_log_syntax_callback(&log_syntax_callback, &s);
```

Comment:
Oh, please drop `&` from before `log_syntax_callback` here. It's a quirk of C that `log_syntax_callback` is equivalent to `&log_syntax_callback` here, but it's confusing and we don't use this style.